bam_sort.c

```diff
@@ -1815,6 +1848,7 @@ int bam_sort_core_ext(int is_by_qname, const char *fn, const char *prefix,
         }
         if (bam_merge_core2(is_by_qname, fnout, modeout, NULL, n_files, fns,
                             MERGE_COMBINE_RG|MERGE_COMBINE_PG|MERGE_FIRST_CO,
+                            //MERGE_FIRST_CO | MERGE_NO_TRANS,
```

Comment:
If we remove that we also want to remove the whole MERGE_NO_TRANS code elsewhere.  I recall working on this in merge so that it was faster and didn't spend a lot of pointless time translating IDs between files when infact all the IDs are identical because we just took the same header and spat it out N times.  This is simply and optimisation to remove that necessity.

Alas what I can't remember is why I didn't enable it.  I'm no expert in merge, so maybe it was just fear of the unknown or maybe it was a work in progress and something was amiss. Hmm.  Can you see anything wrong with the logic of what it's attempting to do?

Is this simply an issue of convolution between samtools merge and samtools sort?